File: src/NTIX.Core/Models/State.cs
namespace NTIX.Core.Models;

public record State(
    int Version = 1,
    Dictionary<string, string>? Winget = null,
    Dictionary<string, string>? Chocolatey = null,
    Dictionary<string, string>? Scoop = null,
    Dictionary<string, string?>? ScoopBuckets = null
)
{
    public Dictionary<string, string> Winget { get; init; } = Winget ?? new();
    public Dictionary<string, string> Chocolatey { get; init; } = Chocolatey ?? new();
    public Dictionary<string, string> Scoop { get; init; } = Scoop ?? new();
    public Dictionary<string, string?> ScoopBuckets { get; init; } = ScoopBuckets ?? new();
}

File: src/models/state.rs
use std::collections::HashMap;

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct State {
    pub version: i32,
    pub winget: HashMap<String, String>,
    pub chocolatey: HashMap<String, String>,
    pub scoop: HashMap<String, String>,
    pub scoop_buckets: HashMap<String, Option<String>>,
    /// Managed config files: dest (absolute path) -> sha256 of the last applied source content.
    pub config_files: HashMap<String, String>,
}

impl Default for State {
    fn default() -> Self {
        Self {
            version: 2,
            winget: HashMap::new(),
            chocolatey: HashMap::new(),
            scoop: HashMap::new(),
            scoop_buckets: HashMap::new(),
            config_files: HashMap::new(),
        }
    }
}
